src/main/java/com/shaft/tools/listeners/XRayListener.java

```diff
@@ -0,0 +1,76 @@
+package com.shaft.tools.listeners;
+
+import com.shaft.tools.tms.XrayIntegrationManager;
+import com.shaft.tools.io.ReportManager;
+import com.shaft.tools.io.ReportManagerHelper;
+import io.qameta.allure.*;
+import org.testng.*;
+
+/**
+ * The listener interface for receiving Xray events.
+ * The Listener will be automatically invoked when TestNG tests are run.
+ * The Listener uses Epic, Feature, Story and Test allure annotations' values.
+ *
+ * @see Allure
+ */
+
+
+public class XRayListener implements IInvokedMethodListener, ITestListener, IExecutionListener  {
```

Comment:
Can we merge this listener with one of the others?

Reply:
Yes, sure. I merged it to AlterSuiteListener, I kept it separated for maintainability.